### PR TITLE
Scale proportionally when the height of the available screen less than 800 px

### PR DESCRIPTION
**Pull request by pzhlkj6612**
# Comparison

### CentOS 8.2.2004 (GNOME 3.32.2)

1368x768

|              | original | modified
| -            | - | -
| list         | list-centos_8 2 2004-1368_768-1-GHA | list-centos_8 2 2004-1368_768-2-proportionally
| settings end | settings_end-centos_8 2 2004-1368_768-1-GHA | settings_end-centos_8 2 2004-1368_768-2-proportionally
| update popup | update_popup-centos_8 2 2004-1368_768-1-GHA | update_popup-centos_8 2 2004-1368_768-2-proportionally

### macOS 11.1.0 20C69

1366x768

|              | original | modified
| -            | - | -
| list         | list-mac11 1 0_20C69-1366_768-1-GHA | list-mac11 1 0_20C69-1366_768-2-proportionally
| settings end | settings_end-mac11 1 0_20C69-1366_768-1-GHA | settings_end-mac11 1 0_20C69-1366_768-2-proportionally
| update popup | update_popup-mac11 1 0_20C69-1366_768-1-GHA | update_popup-mac11 1 0_20C69-1366_768-2-proportionally

### Windows 10 10.0.19042.631

1366x768

|              | original | modified
| -            | - | -
| list         | list-win10-1366_768-1-GHA | list-win10-1366_768-2-proportionally
| settings end | settings_end-win10-1366_768-1-GHA | settings_end-win10-1366_768-2-proportionally
| update popup | update_popup-win10-1366_768-1-GHA | update_popup-win10-1366_768-2-proportionally

### Windows 7 6.1.7600

1366x768

|              | original | modified
| -            | - | -
| list         | list-win7-1366_768-1-GHA | list-win7-1366_768-2-proportionally
| settings end | settings_end-win7-1366_768-1-GHA | settings_end-win7-1366_768-2-proportionally
| update popup | update_popup-win7-1366_768-1-GHA | update_popup-win7-1366_768-2-proportionally

### CODE REVIEW

File: BesWidgets/BesScaleUtil.h [hunk 1]
```diff
@@ -12,12 +12,12 @@ class BesScaleUtil{
         //获取主屏幕可用大小
         QRect primaryScreenRect = getPrimaryScreenRect();
 
-        //窗口本身高800px，分辨率高小于800时缩放到一个固定比例。
+        //窗口本身高800px，分辨率高小于800时，按比例缩放
```
Comment (BensonLaur) [inline BesWidgets/BesScaleUtil.h:15]: 之前提到的窗口本身高 800px 的注释说法现在看来是有错误的

scale 函数不仅用于某些控件的尺寸调整：
https://github.com/BesLyric-for-X/BesLyric-for-X/blob/dcb07041cc300bb7eb9f4ce90b38b7d4b978dc7e/MiddleWidgets/SettingWidget/SuDefaultPath.cpp#L43-L54

主窗口大小本身也是有用到 scale 函数
https://github.com/BesLyric-for-X/BesLyric-for-X/blob/a3be307ac46eb404b3b24b88f371662b70fd9a25/BesWidgets/BesFramelessWidget.cpp#L15-L16

**Review by BensonLaur — APPROVED** [review]

File: BesWidgets/BesScaleUtil.h (hunk 1 above)
Comment (BensonLaur) [reply BesWidgets/BesScaleUtil.h:15]: 虽然后面有提到缩放的问题，不过应该说，“窗口本身高800px” 还是可能会产生一点误解
```
        //分辨率高小于800时，按比例缩放
        //分辨率高大于800，小于1080的情况下不缩放
        //分辨率高大于1080时，按比例缩放
        //比如窗口的基础高度为 800px, 实际的设置高度为 800 * BesScaleUtil::scale()
```

**Comment by pzhlkj6612** [comment]
This change looks terrible. There are not many widgets in the window, but on a 1366x768 desktop, the window occupies the entire desktop. I think we should make it more compact.

Working on it, please wait.

**Comment by pzhlkj6612** [comment]
> This change looks terrible. There are not many widgets in the window, but on a 1366x768 desktop, the window occupies the entire desktop. I think we should make it more compact.

Well, the only way to make the window more compact is to make its content (widgets) more compact. Just adjusting the scaling ratio does not help, because the size of all text is constant (it depends on the value returned by `QApplication::font().pointSize()` , see 5e1edcd6d21db5b922579a23ef9f653e93edcd39 ). In other words, if the widget is too small, it will not be able to hold the text in it.

> Working on it, please wait.

This work will **not** be done soon, because we don't have so much time at the moment.

**Pull request merged**